### ISSUE TITLE
RNA Seq failed exit status 247

### ISSUE THREAD

**Issue by husamia** [google/deepvariant#412]
I thought I should be able to call variants from short reads aligned to the human genome (hg19) with hisat2. So I gave it a try.

Windows 10 Docker desktop. 

`/opt/deepvariant/bin/run_deepvariant --model_type=WGS --output_vcf=CDAR-016_run0124_deepvariant.vcf --reads=CDAR-016_human37_align.bam --ref=./CFM032/human_g1k_v37.fasta --novcf_stats_report --num_shards=5 --call_variants_extra_args "use_openvino=true"`

Error:
```
I0123 08:06:11.771944 139769488508672 make_examples.py:535] Task 1/5: 9704 candidates (9978 examples) [651.40s elapsed]
I0123 08:07:25.050816 140120745805568 make_examples.py:535] Task 3/5: 9400 candidates (9710 examples) [246.46s elapsed]
I0123 08:09:37.633983 140665948894976 make_examples.py:535] Task 4/5: 6501 candidates (6671 examples) [1447.91s elapsed]
I0123 08:10:30.871732 140590877263616 make_examples.py:535] Task 2/5: 7503 candidates (7798 examples) [1754.74s elapsed]
I0123 08:14:47.526611 139645822764800 make_examples.py:535] Task 0/5: 8701 candidates (9015 examples) [4857.13s elapsed]
I0123 08:16:17.660391 140590877263616 make_examples.py:535] Task 2/5: 7600 candidates (7895 examples) [346.79s elapsed]
I0123 08:17:25.425076 140120745805568 make_examples.py:535] Task 3/5: 9501 candidates (9813 examples) [600.37s elapsed]
I0123 08:23:17.762356 140590877263616 make_examples.py:535] Task 2/5: 7700 candidates (8000 examples) [420.10s elapsed]
I0123 08:28:08.231647 140590877263616 make_examples.py:535] Task 2/5: 7800 candidates (8102 examples) [290.47s elapsed]
I0123 08:33:41.777438 139769488508672 make_examples.py:535] Task 1/5: 9800 candidates (10074 examples) [1650.01s elapsed]
parallel: This job failed:
/opt/deepvariant/bin/make_examples --mode calling --ref ./CFM032/human_g1k_v37.fasta --reads run0124_lane9_indexUDI0001-UDI0001=CDAR-016_human37_align.bam --examples /tmp/tmpfh_b9v77/make_examples.tfrecord@5.gz --task 1

real    2480m8.473s
user    2463m44.983s
sys     2m30.474s
I0123 08:51:28.346467 139763485337344 run_deepvariant.py:321] None
Traceback (most recent call last):
  File "/opt/deepvariant/bin/run_deepvariant.py", line 332, in <module>
    app.run(main)
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "/opt/deepvariant/bin/run_deepvariant.py", line 319, in main
    subprocess.check_call(command, shell=True, executable='/bin/bash')
  File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'time seq 0 4 | parallel --halt 2 --line-buffer /opt/deepvariant/bin/make_examples --mode calling --ref "./CFM032/human_g1k_v37.fasta" --reads "run0124_lane9_indexUDI0001-UDI0001=CDAR-016_human37_align.bam" --examples "/tmp/tmpfh_b9v77/make_examples.tfrecord@5.gz" --task {}' returned non-zero exit status 247.

```

**Comment by pichuan** [comment]
@husamia Can you see if any earlier logs have more information? See if you can find another Traceback with more informative error messages?

**Comment by pgrosu** [comment]
@husamia Basically what you are seeing is that you're running out of memory.  Try it on a machine with more memory.

**Comment by husamia** [comment]
> @husamia Can you see if any earlier logs have more information? See if you can find another Traceback with more informative error messages?
I closed the container. 

> @husamia Basically what you are seeing is that you're running out of memory. Try it on a machine with more memory.
that's a good point. is there an option to specify the temp folder outside the container? this would solve my problem.

**Comment by pichuan** [comment]
@husamia I don't think setting temp folder will help with OOM problem. Can you try running with a machine with more RAM, or maybe trying running on a smaller region first (which will not solve the OOM problem but can at least confirm that it's possible to finish a smaller run).

**Comment by husamia** [comment]
> @husamia I don't think setting temp folder will help with OOM problem. Can you try running with a machine with more RAM, or maybe trying running on a smaller region first (which will not solve the OOM problem but can at least confirm that it's possible to finish a smaller run).

I am running it on machine with 64GB of ram of which 80% is used by the Docker. I dont usually have problems with this setup with WGS data.

**Comment by pgrosu** [comment]
@husamia Unfortunately usually is not a reliable description that can be applied to Docker, as it performs many layers of abstraction that can cause a spike when you are not even aware of it.  80% is definitely not enough, especially with the layers of abstractions that DeepVariant also adds.  On top of that you would have the Windows abstraction either running a virtual machine or "leaner" Windows Subsystem for Linux, which has its own abstractions.  Think about it this way, how would you even be able to tell if there is a 10 second spike when Docker starts using 97-98-99% of resources if you need to wait for the OS process resource queue to clear in order to check the Docker status, but are limited/prevented to do so by multiple layers of resource management systems?

**Comment by pichuan** [comment]
Thanks @pgrosu for the help!
@husamia Feel free to open another issue if you have further questions. I'll close this now.

**Comment by danielecook** [comment]
@husamia although this was closed some time ago, we have just released an Illumina RNA-seq model and [case study](https://github.com/google/deepvariant/blob/r1.4/docs/deepvariant-rnaseq-case-study.md). If you are still interested in calling variants from RNA-seq data using DeepVariant, this should work for you.

We have also updated the DeepVariant code base to be more memory efficient with RNA-seq data. This involves passing a new flag (`--split_skip_reads`), that allows for reads containing large SKIP to be processed efficiently.
